### PR TITLE
Migrated the restart documentation

### PR DESCRIPTION
**Pull request by CaroleAnneDaunais**
# Description of the problem

Transfered and updated the Restart documentation

### CODE REVIEW

File: doc/source/parameters/cfd/restart.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Finite element interpolation (FEM)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This subsection controls the restart and checkpointing of the simulation. 
+This feature is used to restart a time-dependant simulation using the data written on the last checkpoint before the smulation stopped or crashed.
```
Comment (jeannejoachim) [inline doc/source/parameters/cfd/restart.rst:5]: before the simulation stopped

File: doc/source/parameters/cfd/restart.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+.. code-block:: text
+
+  subsection restart
+    set checkpoint = false
+    set restart    = false
+    set filename   = restart
+    set frequency  = 1
+  end
+
+* The ``checkpoint`` option enables checkpointing simulation informations when it is set to ``true``. You may want to enable it when launching a time-dependant simulation if you fear you might either run out of memory/simulation time during the simulation.
```
Comment (jeannejoachim) [inline doc/source/parameters/cfd/restart.rst:17]: creating a checkpoint of the simulation?

Comment (jeannejoachim) [reply doc/source/parameters/cfd/restart.rst:17]: put "You may want to enable it when launching a time-dependant simulation if you fear you might either run out of memory/simulation time during the simulation." in a tip box

File: doc/source/parameters/cfd/restart.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+* The ``restart`` option controls if the simulation will be restarted from a previous checkpoint. Turn this parameter to ``false`` for the first checkpoint creation. The simulation will start from the last checkpoint when the parameter is set to ``true`` and will finish at the provided time end (see `Simulation Control <https://lethe-cfd.github.io/lethe/parameters/cfd/simulation_control.html>`).
```
Comment (jeannejoachim) [inline doc/source/parameters/cfd/restart.rst:19]: do not forget the _ at the end of a html link:
Simulation Control <https://lethe-cfd.github.io/lethe/parameters/cfd/simulation_control.html>`_

File: doc/source/parameters/cfd/restart.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+* The ``filename`` parameter fixes the prefix of all the checkpointing files. 
+
+.. warning::
+
+  Restart files are searched for in the output path specified in subsection simulation control.
+
+* The ``frequency`` parameter controls the checkpointing frequency. Checkpointing is very intensive from an I/O point of view, consequently it should not be done at every iteration. The checkpointing frequency is is accordance to the number of ``time step`` (see `Simulation Control <https://lethe-cfd.github.io/lethe/parameters/cfd/simulation_control.html>`).
```
Comment (jeannejoachim) [inline doc/source/parameters/cfd/restart.rst:27]: put in a tip box the part (+ fix html link):
The checkpointing frequency is is accordance to the number of ``time step`` (see `Simulation Control <https://lethe-cfd.github.io/lethe/parameters/cfd/simulation_control.html>_`).

**Review by jeannejoachim — COMMENTED** [review]
Clean! A few changes, overwise very straight forward, well done

**Review by jeannejoachim — APPROVED** [review]
Perfect thank you!

File: doc/source/parameters/cfd/restart.rst [hunk 5]
```diff
@@ -0,0 +1,30 @@
+Finite element interpolation (FEM)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This subsection controls the restart and checkpointing of the simulation. 
+This feature is used to restart a time-dependant simulation using the data written on the last checkpoint before the simulation stopped or crashed.
```
Comment (lpsaavedra) [inline doc/source/parameters/cfd/restart.rst:5]:
```suggestion
This feature is used to restart a time-dependent simulation using the data written on the last checkpoint before the simulation stopped or crashed.
```

File: doc/source/parameters/cfd/restart.rst [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    set checkpoint = false
+    set restart    = false
+    set filename   = restart
+    set frequency  = 1
+  end
+
+* The ``checkpoint`` option enables creating a checkpoint of the simulation when it is set to ``true``. 
+
+.. tip::
+  You may want to enable it when launching a time-dependant simulation if you fear you might either run out of memory/simulation time during the simulation.
```
Comment (lpsaavedra) [inline doc/source/parameters/cfd/restart.rst:20]:
```suggestion
  You may want to enable it when launching a time-dependent simulation if you fear you might either run out of memory/simulation time during the simulation.
```

**Review by lpsaavedra — APPROVED** [review]
Nice work! Only found one typo twice but otherwise I think it is very clear

**Pull request merged**